Userland/Libraries/LibC/malloc.cpp

```diff
@@ -25,9 +25,14 @@ class PthreadMutexLocker {
     ALWAYS_INLINE explicit PthreadMutexLocker(pthread_mutex_t& mutex)
         : m_mutex(mutex)
     {
+        __heap_is_stable = false;
```

Comment:
These need to happen under the lock scope? Otherwise a thread getting preemepted, etc could alter the execution ordering and corrupt the state?